server.go

```diff
@@ -3126,3 +3126,50 @@ func (s *server) watchChannelStatus() {
 		}
 	}
 }
+
+// isNodePublic attempts to determine if a node with the given public key is
+// publicly advertised within the network.
+func (s *server) isNodePublic(pubKey *btcec.PublicKey) (bool, error) {
+	// First, we'll attempt to retrieve the node from the graph.
+	remoteNode, err := s.chanDB.ChannelGraph().FetchLightningNode(pubKey)
```

Comment:
Isn't this enough? In that if the node isn't found in our graph, then we can assume that it likely has purely non-advertised channels. For example, here's me trying to lookup laptop node (running the app on testnet) from the faucet: 
```
kek@segspam:~/gocode/src/github.com/lightningnetwork/lnd# tlncli getnodeinfo 028916c993156097a1fdb5b4bbe4116f63bf9afbf4c7210d1c638dd2127abe7ed4
tlncli: command not found
```

Reply:
Ah actually this doesn't work if a direct connection exists between the two endpoints.